lib/shard/src/segment_holder/mod.rs

```diff
@@ -295,6 +295,16 @@ impl SegmentHolder {
         self.non_appendable_segments.keys().copied().collect()
     }
 
+    /// Bumps the version of a random appendable segment.
+    pub fn bump_version_of_random_appendable(&self, op_num: SeqNumberType) {
+        if let Some(segment) = self.random_appendable_segment() {
+            match segment {
+                LockedSegment::Original(rw_lock) => rw_lock.write().bump_segment_version(op_num),
+                LockedSegment::Proxy(rw_lock) => rw_lock.write().bump_version(op_num),
+            }
+        }
+    }
```

Comment:
I _think_ it may be better to pick a segment through `self.aloha_random_write()` and supply `self.appendable_segments_ids()`.

That way we're sure we're getting a segment with write access. It may prevent blocking for a long time if some long running write operation is going on on the random segment we've selected.